hypershift-operator/controllers/nodepool/openstack.go

```diff
@@ -57,16 +57,15 @@ func (r *NodePoolReconciler) setOpenStackConditions(ctx context.Context, nodePoo
 // An ORC object will be created or updated with the image spec.
 // The image name will be returned.
 func (r *NodePoolReconciler) reconcileOpenStackImageCR(ctx context.Context, client client.Client, hcluster *hyperv1.HostedCluster, release *releaseinfo.ReleaseImage, nodePool *hyperv1.NodePool) (string, error) {
-	releaseVersion, err := openstack.OpenStackReleaseImage(release)
+	controlPlaneNamespace := fmt.Sprintf("%s-%s", nodePool.Namespace, strings.ReplaceAll(nodePool.Spec.ClusterName, ".", "-"))
+	openstackCluster, err := openstack.GetOpenStackClusterForHostedCluster(ctx, client, hcluster, controlPlaneNamespace)
 	if err != nil {
 		return "", err
 	}
-	controlPlaneNamespace := fmt.Sprintf("%s-%s", nodePool.Namespace, strings.ReplaceAll(nodePool.Spec.ClusterName, ".", "-"))
-	openstackCluster, err := openstack.GetOpenStackClusterForHostedCluster(ctx, client, hcluster, controlPlaneNamespace)
+	imageName, err := openstack.ClusterImageName(hcluster, release)
```

Comment:
You don't actually need to change `imageName` here. You can leave it as before. This is the name of the ORC image, which is already namespaced (because it's literally in a kubernetes namespace).

ReconcileOpenStackImageSpec is setting the name of the glance image, which is what needs to be prefixed with the hosted cluster.

This is ok, I guess, if you think it's simpler. It just struck me as weird that we were calling ClusterImageName() twice.

Reply:
But we could have multiple Hosted Clusters within the same namespace, which is why I added it here.

Reply:
Since the ORC resource is in the HC namespace, we can use the previous name, gotcha.